Tests/WeaverTests.cs

```diff
@@ -156,7 +170,18 @@ public void UniquePropertyTypeButExplicitMethodName_WithIsEmittedWithLongName()
     public void OriginalWithMethodIsRemoved()
     {
         var type1 = assembly.GetType("AssemblyToProcess.PrimitiveValues");
-        Assert.False(type1.GetMethods().Any(m => m.Name == "With" && m.IsGenericMethod));
+        // check for the generic With method
+        Assert.False(type1.GetMethods().Any(m =>
+            m.IsPublic &&
+            m.Name == "With" &&
+            m.IsGenericMethod &&
+            m.GetParameters().Length == 1));
+        // check for a With method with Object parameter
+        Assert.False(type1.GetMethods().Any(m => 
+            m.IsPublic &&
+            m.Name == "With" && 
+            m.GetParameters().Length == 1 &&
+            m.GetParameters()[0].ParameterType == typeof(object)));
```

Comment:
While testing the code for this PR, I noticed most test cases had an extra With. I thought RemoveGenericWith() was broken so I starting rewriting it. I then realised the issue was that most test cases have
```csharp
public MultipleConstructors With(object value) => this;
```
instead of the generics suggested in the documentation.
I first thought this was a valid alternative but now I realized that it may be an issue for classes with properties of type System.Object.
I'm removing this portion of the code but the test cases should be fixed.
instead of the